extensions/standard-processors/processors/PutFile.cpp

```diff
@@ -54,6 +54,19 @@ core::Property PutFile::CreateDirs("Create Missing Directories", "If true, then
 core::Property PutFile::MaxDestFiles(
     core::PropertyBuilder::createProperty("Maximum File Count")->withDescription("Specifies the maximum number of files that can exist in the output directory")->withDefaultValue<int>(-1)->build());
 
+#ifndef WIN32
+core::Property PutFile::Permissions(
+    core::PropertyBuilder::createProperty("Permissions")
+      ->withDescription("Sets the permissions on the output file to the value of this attribute. "
+                        "Format must be format octal number (e.g. 644 or 0755). Not supported on Windows systems.")
+      ->build());
+core::Property PutFile::DirectoryPermissions(
+    core::PropertyBuilder::createProperty("Directory Permissions")
+      ->withDescription("Sets the permissions on the directories being created if 'Create Missing Directories' property is set. "
+                        "Format must be format octal number (e.g. 644 or 0755). Not supported on Windows systems.")
+      ->build());
```

Comment:
What happens if these are not set?  I would have expected a default value here.

Reply:
In that case it works the same way as `boost::filesystem::create_directory` which is creating the directory in mode `777` then applies the system's default umask.